lxd/instance/drivers/driver_qemu.go

```diff
@@ -1127,13 +1162,70 @@ func (d *qemu) Start(stateful bool) error {
 	// Reset timeout to 30s.
 	op.Reset()
 
+	// Restore the state.
+	if stateful {
+		stateFile, err := os.Open(d.StatePath())
```

Comment:
@stgraber this is just a personal code style thing, but I would consider moving this step into its own unexported function just to separate the logic step (restoring the state) from non-trivial implementation (all the pipe work), so that a reader only needs to drill down into it if they are interested. As `Start` is getting quite long and tricky to reason about.

Reply:
Yeah, we can do that. I'll keep the logic in the Stop function inline for now though and will split it out when I do snapshots.